ec2/auto_scaling_group/templates/scripts/asg_user_data.sh

```
@@ -0,0 +1,494 @@
+log "  - Instance ID: $(curl -s http://169.254.169.254/latest/meta-data/instance-id)"
+log "  - Instance Type: $(curl -s http://169.254.169.254/latest/meta-data/instance-type)"
+log "  - Hostname: $(hostname)"
+log "  - Private IP: $(curl -s http://169.254.169.254/latest/meta-data/local-ipv4)"
+log "  - ASG Name: ${asg_name}"
+log "  - Region: $AWS_REGION"
+
+# 起動完了の通知
+log "Auto Scaling Group instance initialization completed successfully"
+
+# CloudWatch メトリクス送信（オプション）
+%{ if enable_cloudwatch_metrics ~}
```

Shellcheck (0.10.0)

[warning] This is interpreted as a command name ending with '{'. Double check syntax.

(SC2288)